### DISCUSSION TITLE
Generate release notes with comparison of consumed artifacts

### DISCUSSION THREAD

**Discussion by ErikTempelaarVO** [rfennell/AzurePipelines#1553] [Q&A] [answered]
The project I would like to generate release notes for is a combined artifact.
Its pipeline gathers several artifacts from other pipelines' master builds and combines these components into a single archive.
And this is when I want to create a list of work items, and possibly provide another build to compare against (manually).
(We create sprint-releases, for a single iteration, and stable-releases, that span several iterations that must list all changes)
A single component is unaware of the type of release it will end up in and can't do any comparisons on its own.

I think it should go something like this for a new release:
- Gather consumed artifacts for the current build
- Gather consumed artifacts for the old build to compare against
- For each consumed artifact, run a comparison of the current build to the old and check for merged PRs and links to work items etc.

It seems the 'Related work items' for a consumed artifact only lists the most recent change that went into master.
I'm looking for a list of work items since some earlier build.

Can this project help with this?

**Comment by rfennell** [comment, marked as answer]
The older 'Classic Release' functionality of my task sort of does a gather of associations, but as you said it only get the details of the associated items with the consumed build artifact in a simplistic way. But anyway, as multistage YAML is the future, even this partial solution is of little use and I don't plan any changes to the 'Classic Build  or 'Classic Release' functionality in the task.

The way I would tackle this problem is to use my task in each consumed build pipeline, so they can create release notes on the changes since the last successful build (so this could branch, stage, PR etc. based). I would publish this markdown document as a pipeline artifact and then the main release pipeline use a script to merge these various release note markdown fragments into the required document.
